transport_plugins/bled112/iotile_transport_bled112/bled112_cmd.py

```diff
@@ -299,11 +326,9 @@ def notified_payload(event):
 
         supported_auth = AuthType.AUTH_METHOD_0.value | AuthType.AUTH_METHOD_1.value \
                          | AuthType.AUTH_METHOD_2.value | AuthType.AUTH_METHOD_3.value
-        permisions = 0x00
-        token_gen = 0x01
 
-        manager = BLED112AuthManager(permisions, token_gen)
-        success, data = manager.authenticate(device_uuid, supported_auth, send_auth_client_request)
+        self._auth_manager._key_provider.conn_map = self._conn_map
```

Comment:
Gives the connection map over to the authentication chain so that it can look for device aliases if needed.

Reply:
In terms of what part of the system does what, I would recommend refactoring this:
- the auth system is designed to work via device ids, not connection strings / mac addresses so I would not leak the connection map over to that system.
- instead, you can just do the lookup inside the bled112 code between `connection_string` and `device_id` and then only ever hand a `device_id` over to the auth system so it doesn't need to worry about it.
- if you can't find the right device id in your conn_map then it fails the authentication procedure without asking `_auth_manager` for anything since it doesn't have the right prerequisites (no device id).